starter_kits/C/src/MyBot.c

```diff
@@ -0,0 +1,324 @@
+unsigned char dropoffs[256][2], extract_ratio, inspiration_enabled, inspiration_radius, inspiration_ship_count, inspired_extract_ratio, inspired_bonus_multiplier, inspired_move_cost_ratio, mhh, mhw, move_cost_ratio, my_id, map_size[2], number_of_players;
+unsigned short dropoff_cost, max_energy, max_turns, ship_cost, turn_number;
+
+FILE *log_file = NULL;
+
+
+
+// Now let's declare some helper functions
+// You might reasonably want to keep some of these
+
+// Normalize X value (map_width + 1 = 0, 0 - 1 = map_width)
+inline unsigned char norm_x(char x){
+	return (x + map_size[0]) % map_size[0];
+}
+
+// Normalize Y value (map_height + 1 = 0, 0 - 1 = map_height)
+inline unsigned char norm_y(char y){
+	return (y + map_size[1]) % map_size[1];
+}
+
+// Calculate the distance betwen two sets of x and y values
+inline unsigned char calc_dist(unsigned char ax, unsigned char ay, unsigned char bx, unsigned char by){
+	unsigned char dx, dy;
+
+	if(ax > bx)
+		dx = ax - bx;
+	else
+		dx = bx - ax;
+
+	if(ay > by)
+		dy = ay - by;
+	else
+		dy = by - ay;
+
+	return (dx % mhw) + (dy % mhh);
+}
+
+// 0 if not occupied, 1 if occupied
+inline unsigned char tile_occupied(unsigned char x, unsigned char y){
+	return map[x][y].occupied != (unsigned short)-1;
+}
+
+// Now for the commands to send to the game engine
+inline void move_still(unsigned short ship_id){
+	printf("m %hu o ", ship_id);
+}
+
+inline void move_north(unsigned short ship_id){
+	printf("m %hu n ", ship_id);
+}
+
+inline void move_south(unsigned short ship_id){
+	printf("m %hu s ", ship_id);
+}
+
+inline void move_east(unsigned short ship_id){
+	printf("m %hu e ", ship_id);
+}
+
+inline void move_west(unsigned short ship_id){
+	printf("m %hu w ", ship_id);
+}
+
+
+inline void convert_dropoff(unsigned short ship_id){
+	printf("c %hu ", ship_id);
+}
+
+inline void spawn_ship(){
+	printf("g ");
+}
+
+
+
+// Now that all of the variables have been declared, we can write some basic "smart" logic
+void random_move(unsigned short ship_id){
+	unsigned char decision = rand() % 100;
+
+	// 10% chance each to move north/east/south/west (This is extra smart)
+	if(decision <= 10){
+		move_north(ship_id);
+	}else if(decision <= 20){
+		move_east(ship_id);
+	}else if(decision <= 30){
+		move_south(ship_id);
+	}else if(decision <= 40){
+		move_west(ship_id);
+	}else{
+		// And a 60% chance to stay still (This is also extra smart)
+		// We could alternatively just not give the ship a move this turn
+		move_still(ship_id);
+	}
+}
+
+
+// There's something that random_move is forgetting... Let's avoid collisions!
+void naive_navigate(unsigned short ship_id){
+	unsigned char decision = rand() % 100, x = ships[ship_id].x, y = ships[ship_id].y;
+
+	// North is y - 1
+	// South is y + 1
+	// West is x - 1
+	// East is x + 1
+
+	// 10% chance each to move north/east/south/west (This is even smarter than before!)
+	if(decision <= 10 && tile_occupied(x, norm_y(y - 1)) == 0){
```

Comment:
These if statements should be 1 line functions.  Too complicated and not obvious.

Reply:
I don't understand what's not obvious about it...? Maybe I could check if each of the tiles are occupied outside of the if statements, put the results in 4 variables, and then check the variables in the if statement instead of calling the functions directly, but I think anybody who's not brand-new to programming could tell what this does. (And if someone is brand-new to programming, let's be real, they're going to use the Python starter kit)

Reply:
1) If you have to leave a comment in the code, that’s a sign the readability isn’t on point.
2) It is a starter kit ;)